framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -546,14 +651,35 @@ private String getCharArrayStringVal(List<? extends ExpressionTree> initializers
         }
 
         @Override
-        public Void visitTypeCast(TypeCastTree tree, AnnotatedTypeMirror type) {
-            if (handledByValueChecker(type)) {
-                AnnotatedTypeMirror castedAnnotation = getAnnotatedType(tree.getExpression());
-                List<?> values = getValues(castedAnnotation, type.getUnderlyingType());
-                type.replaceAnnotation(createResultingAnnotation(type.getUnderlyingType(), values));
-            } else if (type.getKind() == TypeKind.ARRAY) {
+        public Void visitTypeCast(TypeCastTree tree, AnnotatedTypeMirror atm) {
+            if (handledByValueChecker(atm)) {
+                AnnotationMirror oldAnno =
+                        getAnnotatedType(tree.getExpression()).getAnnotationInHierarchy(UNKNOWNVAL);
+                if (oldAnno != null) {
+                    TypeMirror newType = atm.getUnderlyingType();
+                    AnnotationMirror newAnno;
+                    Range range;
+                    if (AnnotationUtils.areSameByClass(oldAnno, IntRange.class)
+                            && (range = getIntRange(oldAnno)).isWiderThan(MAX_VALUES)) {
```

Comment:
It's not really nice that there is an assignment within the condition. You could limit the scope of range to within the first condition and have a separate if. Or would that change require duplicating the outer else branch?